File: src/main/java/com/bourgeolet/task_manager/dto/task/TaskCreateDTO.java
package com.bourgeolet.task_manager.dto.task;

import com.bourgeolet.task_manager.dto.tag.TagDTO;
import jakarta.validation.constraints.NotBlank;

import java.util.List;

public record TaskCreateDTO(
        @NotBlank String actor,

        String userAffectee,
        @NotBlank String title,
        String description,
        int priority,
        List<TagDTO> tags
) {
}

File: frontend/src/app/features/task/models/task-create-dto.ts
import { TagCreateDto } from '../../tag/models/tag-create-dto';

export interface TaskCreateDto {
  actor: string;
  title: string;
  description: string | null | undefined;
  userAffectee: string | null | undefined;
  priority: number | null | undefined;
  tags: TagCreateDto[] | null | undefined;
}
